### PR TITLE
Merging Packaging projects

### PR DESCRIPTION
**Pull request by emgarten**
* Merging NuGet.Packaging.Core.Types into NuGet.Packaging.Core
* Merging NuGet.ContentModel, NuGet.Client, and NuGet.RuntimeModel into NuGet.Packaging
* Merging NuGet.Repositories into NuGet.Protocol

NuGet.Packaging contains everything needed for reading runtime.json from nupkgs and the content model for finding v3 assets.

NuGet.Packaging.Core is still a separate project. It is a light weight zip/nuspec reader that can find the id/version of a package and read basic nuspec properties. And without all the support for pack and the extra dependency on newtonsoft.json. 

NuGet.Packaging.Core could go into NuGet.Packaging also, but for now I'm leaving it out unless someone has a strong opinion that this shouldn't exist.

This change removes 5 projects and 3 test projects for the solution.

### CODE REVIEW

File: src/NuGet.Clients/NuGet.Credentials/NuGet.Credentials.csproj [hunk 1]
```diff
@@ -17,13 +17,11 @@
...
   </ItemGroup>
+
   <ItemGroup>
-    <ProjectReference Include="$(NuGetCoreSrcDirectory)\NuGet.PackageManagement\NuGet.PackageManagement.csproj">
-      <Project>{394aeb96-493c-4839-a5ac-8d93cd2fad40}</Project>
-      <Name>NuGet.PackageManagement</Name>
-    </ProjectReference>
+    <ProjectReference Include="$(NuGetCoreSrcDirectory)\NuGet.PackageManagement\NuGet.PackageManagement.csproj" />
   </ItemGroup>
-  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
```
Comment (mishra14) [inline src/NuGet.Clients/NuGet.Credentials/NuGet.Credentials.csproj:26]: why is this no longer needed?

File: src/NuGet.Core/NuGet.Packaging/NuGet.Packaging.csproj [hunk 2]
```diff
@@ -4,7 +4,7 @@
   <PropertyGroup>
     <Description>NuGet's implementation for reading nupkg package and nuspec package specification files.</Description>
     <TargetFrameworks>net45;netstandard1.3</TargetFrameworks>
-    <NoWarn>$(NoWarn);CS1591;CS1574;CS1573</NoWarn>
+    <NoWarn>$(NoWarn);CS1591;CS1574;CS1573;CS1572</NoWarn>
```
Comment (mishra14) [inline src/NuGet.Core/NuGet.Packaging/NuGet.Packaging.csproj:7]: Looking at https://msdn.microsoft.com/en-us/library/hba91y19(v=vs.140).aspx. Why do we want to ignore these?

File: src/NuGet.Core/NuGet.Packaging/NuGet.Packaging.csproj [hunk 3]
```diff
@@ -14,7 +14,7 @@
     <ProjectReference Include="..\NuGet.Common\NuGet.Common.csproj" />
     <ProjectReference Include="..\NuGet.Packaging.Core\NuGet.Packaging.Core.csproj" />
   </ItemGroup>
-
+
```
Comment (mishra14) [inline src/NuGet.Core/NuGet.Packaging/NuGet.Packaging.csproj:17]: empty space.

File: test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/NuGet.PackageManagement.VisualStudio.Test.csproj [hunk 4]
```diff
@@ -35,25 +35,11 @@
...
-      <Name>NuGet.PackageManagement.VisualStudio</Name>
-    </ProjectReference>
-    <ProjectReference Include="..\..\..\src\NuGet.Clients\PackageManagement.UI\NuGet.PackageManagement.UI.csproj">
-      <Name>NuGet.PackageManagement.UI</Name>
-    </ProjectReference>
-    <ProjectReference Include="..\..\..\src\NuGet.Clients\VisualStudio.Facade\NuGet.VisualStudio.Facade.csproj">
-      <Project>{EEA49A74-6EFC-410E-9745-BAD367AC151D}</Project>
-      <Name>NuGet.VisualStudio.Facade</Name>
-    </ProjectReference>
+    <ProjectReference Include="..\..\..\src\NuGet.Clients\NuGet.Credentials\NuGet.Credentials.csproj" />
```
Comment (mishra14) [inline test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/NuGet.PackageManagement.VisualStudio.Test.csproj:38]: Can we add a `$(NuGetClientsSrcDirectory)` as well? that will make all the references cleaner. thoughts?

**Review by mishra14 — COMMENTED** [review]
take a look at the comments

File: src/NuGet.Clients/NuGet.Credentials/NuGet.Credentials.csproj (hunk 1 above)
Comment (mishra14) [reply src/NuGet.Clients/NuGet.Credentials/NuGet.Credentials.csproj:26]: Discussed offline. This is implied by .Net SDK.

Comment (emgarten) [reply src/NuGet.Clients/NuGet.Credentials/NuGet.Credentials.csproj:26]: netcore sdk imports it. the build was giving duplicate import warnings.

File: src/NuGet.Core/NuGet.Packaging/NuGet.Packaging.csproj (hunk 2 above)
Comment (emgarten) [reply src/NuGet.Core/NuGet.Packaging/NuGet.Packaging.csproj:7]: It can be removed when the docs are fixed. The other project was ignoring it.

Comment (emgarten) [reply src/NuGet.Core/NuGet.Packaging/NuGet.Packaging.csproj:7]: Tracked in https://github.com/NuGet/Home/issues/2537

File: test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/NuGet.PackageManagement.VisualStudio.Test.csproj (hunk 4 above)
Comment (emgarten) [reply test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/NuGet.PackageManagement.VisualStudio.Test.csproj:38]: I think we should. It needs to be done for all projects. I've just been incrementally making improvements every time I go in a file.

**Review by mishra14 — APPROVED** [review]
Looks good!

**Pull request closed without merging**